diff --git a/.dockerignore b/.dockerignore
new file mode 100644
index 0000000..ed215cc
--- /dev/null
+++ b/.dockerignore
@@ -0,0 +1,6 @@
+.git*
+.travis*
+.idea/
+*.iml
+README.md
+target/
diff --git a/Dockerfile b/Dockerfile
new file mode 100644
index 0000000..b8e1d85
--- /dev/null
+++ b/Dockerfile
@@ -0,0 +1,15 @@
+FROM openjdk:9-slim
+
+MAINTAINER David Dooling <david@atomist.com>
+
+RUN mkdir -p /opt/app
+
+WORKDIR /opt/app
+
+EXPOSE 8080
+
+CMD ["/usr/bin/java", "-Xmx512m", "-Djava.security.egd=file:/dev/urandom", "-jar", "spring-boot.jar"]
+
+COPY . .
+
+RUN ./mvnw -B package && cp target/*.jar spring-boot.jar && ./mvnw -B clean && rm -rf .mvn mvn* pom.xml src $HOME/.m2
